src/Components/Legend.jsx

```diff
@@ -0,0 +1,137 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import { Segment, Sidebar, Grid, Menu, Sticky, Icon, Header } from 'semantic-ui-react'
+
+class Legend extends Component {
+
+  hideLegend = async () => {
+    await this.props.dispatch({ type: 'CHANGE_LEGEND_VISIBILITY' })    
+  }
+
```

Comment:
out of curiosity, since we only have one action happening here, why we need `async-await`?